h2/src/main/org/h2/command/Parser.java

```diff
@@ -8013,6 +8013,15 @@ private void parseTableColumnDefinition(CommandWithColumns command, Schema schem
             if (NullConstraintType.NULL_IS_NOT_ALLOWED == parseNotNullConstraint()) {
                 column.setNullable(false);
             }
+            String comment = readCommentIf();
+            if (comment != null) {
+                if (column.getComment() != null) {
```

Comment:
These two conditions should be simply swapped.

`readCommentIf()` should be invoked only when comment was not already set.